src/control/server/harness.go

```diff
@@ -228,6 +231,61 @@ func (h *IOServerHarness) startInstances(ctx context.Context, membership *system
 	return nil
 }
 
+// StopInstances will stop harness-managed instances.
+//
+// Iterate over instances and call Stop(signal) on each, return when all
+// instances are stopped or context is done.
+func (h *IOServerHarness) StopInstances(log logging.Logger, ctx context.Context, signal os.Signal, ranks ...uint32) error {
+	if !h.IsStarted() {
```

Comment:
It would be nice to instrument these start/stop methods as well, to allow us to more easily identify performance regressions in time-sensitive functions. That's probably best handled in another PR, though.